python_modules/dagster/dagster/_core/definitions/job_definition.py

```diff
@@ -328,38 +328,62 @@ def dependencies(self) -> DependencyMapping[NodeInvocation]:
     @public
     @property
     def executor_def(self) -> ExecutorDefinition:
+        """Returns the default :py:class:`ExecutorDefinition` for the job.
+
+        If the user has not specified an executor definition, this will be the :py:func:`multi_or_in_process_executor`, which defaults to multiprocess mode.
+        """
         return self._executor_def or DEFAULT_EXECUTOR_DEF
 
     @public
     @property
     def has_specified_executor(self) -> bool:
+        """Returns True if this job has explicitly specified an executor, and False if the executor was inherited through defaults or the :py:class:`Definitions` object the job was provided to.
+        """
         return self._executor_def is not None
 
     @public
     @property
     def resource_defs(self) -> Mapping[str, ResourceDefinition]:
+        """Returns the set of ResourceDefinition objects specified on the job.
+
+        This may not be the complete set of resources required by the job, since those can also be provided on the :py:class:`Definitions` object the job may be provided to.
+        """
         return self._resource_defs
 
     @public
     @property
     def partitioned_config(self) -> Optional[PartitionedConfig]:
+        """The partitioned config for the job, if it has one.
+
+        A partitioned config defines a way to map partition keys to run config for the job.
+        """
         return self._partitioned_config
 
     @public
     @property
     def config_mapping(self) -> Optional[ConfigMapping]:
+        """The config mapping for the job, if it has one.
+
+        A config mapping defines a way to map a top-level config schema to run config for the job.
+        """
         return self._config_mapping
 
     @public
     @property
     def loggers(self) -> Mapping[str, LoggerDefinition]:
+        """Returns the set of LoggerDefinition objects specified on the job.
+
+        If no loggers were provided on the job, this will be the :py:func:`colored_console_logger`.
+        """
```

Comment:
Aren't these inherited from `Definitions` just like the executor? If so, I would use similar language to my executor comment.

Reply:
Good point regarding inherited from Definitions - although if nothing is specified on Definitions, a default is still set.

Reply:
We should probably change it so that `Definitions` just has a default, that simplifies the model and makes it so that the job inherits from the `Definitions` object with no caveat.

Reply:
I think that change will be hairier than we anticipate (as in, a few hours of work rather than a one liner). For now, I think we should describe the situation as accurately as possible.